src/cls/IPM/Repo/Oras/PackageService.cls

```diff
@@ -56,6 +60,115 @@ Method HasModule(pModuleReference As %IPM.Storage.ModuleInfo) As %Boolean
 	Return ..HasModulePy(..Location, name, ..Namespace, client)
 }
 
+/// Given a list of ORAS tags, aggregate the platform versions by package version
+/// e.g. $lb("1.0.0__2024.1, "1.0.0__2024.2", "2.0.0__2023.1", "3.0.0") -> aggregatedPlatformVersion
+/// where aggregatedPlatformVersion("1.0.0") = $lb("2024.1","2024.2") 
+///       aggregatedPlatformVersion("2.0.0") = $lb("2023.1")
+///       aggregatedPlatformVersion("3.0.0") = $lb("")
+/// The returned list represents the correct order of the package versions, where latest versions are first
+/// e.g., $lb("1.0.0", "2.0.0", "3.0.0") -> $lb("3.0.0", "2.0.0", "1.0.0")
+ClassMethod AggregatePlatformVersions(tagsList As %List, Output aggregatedPlatformVersion As %String) As %List
+{
+	Set ptr = 0
+	Set sorter = ""
+	Set order = ""
+	While $ListNext(tagsList, ptr, tag) {
+		// aggregate
+		Set $ListBuild(version, platformVersion) = $ListFromString(tag, $$$OrasTagPlatformSeparator)
+		Set value = $Get(aggregatedPlatformVersion(version))
+		Set aggregatedPlatformVersion(version) = value _ $ListBuild(platformVersion) 
+
+		// prepare for sorting
+		Set semver = ##class(%IPM.General.SemanticVersion).FromString($$$Tag2Semver(version))
+		Set sorter(+semver.Major, +semver.Minor, +semver.Patch, " "_semver.Prerelease, " "_semver.Build) = version
+	}
+
+	Set major = ""
```

Comment:
This is a great use case for [$query](https://docs.intersystems.com/iris20243/csp/docbook/Doc.View.cls?KEY=RCOS_fquery) but don't need to do that right now.